From: Hannes Reinecke <hare@suse.de>
Date: Fri, 20 Sep 2019 09:56:59 +0200
Subject: [PATCH] block: kABI fixes for BLK_EH_DONE renaming
Patch-Mainline: never, SLE15 kABI fix
References: bsc#1142076

We can't just rename an exported enum variable; we'll rather have
to keep the old value and make the new one an alias to it.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 include/linux/blkdev.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index e694692ca968..98cab4936f29 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -296,11 +296,13 @@ typedef int (init_rq_fn)(struct request_queue *, struct request *, gfp_t);
 typedef void (exit_rq_fn)(struct request_queue *, struct request *);
 
 enum blk_eh_timer_return {
-	BLK_EH_DONE,
+	BLK_EH_NOT_HANDLED,
 	BLK_EH_HANDLED,
 	BLK_EH_RESET_TIMER,
 };
 
+#define BLK_EH_DONE BLK_EH_NOT_HANDLED
+
 typedef enum blk_eh_timer_return (rq_timed_out_fn)(struct request *);
 
 enum blk_queue_state {
-- 
2.16.4

